apps/glusterfs/app_block_volume.go

```diff
@@ -141,6 +141,9 @@ func (a *App) BlockVolumeInfo(w http.ResponseWriter, r *http.Request) {
 		if err != nil {
 			http.Error(w, err.Error(), http.StatusInternalServerError)
 			return err
+		} else if !entry.Visible() {
+			http.Error(w, "Id not found", http.StatusNotFound)
+			return nil
```

Comment:
i think we need to return ErrNotFound

Reply:
I will double check it.

Reply:
1.) it should treat the !visible condition just like not finding the entry. hence we need to return ErrNotFound
2.) In this case (as opposed to the volumeinfo function), the location is wrong. needs to in the block line 132 ff